Model/Sales/Repository/PackageRepository.php

```diff
@@ -94,39 +112,48 @@ public function productWithoutDeliveryOptions(array $products)
     }
 
     /**
+     * @param $product
+     * @param $mailBox
+     *
      * @return bool
      */
-    public function fitInMailbox(): bool
+    public function fitInMailbox($product, $mailBox): bool
     {
+        $mailboxPercentage    = $this->getMailboxPercentage();
+        $mailboxPercentage    += $mailBox * $product->getQty();
+        $maximumMailboxWeight = $this->getWeightTypeOfOption($this->getMaxMailboxWeight());
+        $orderWeight          = $this->getWeightTypeOfOption($this->getWeight());
         if (
-            $this->getCurrentCountry() !== 'NL' ||
-            ! $this->isMailboxActive() ||
-            ! $this->isAllProductsFitInMailbox() ||
-            ! $this->getWeight() ||
-            $this->getMailboxPercentage() < 100 &&
-            $this->getWeight() > $this->getMaxWeight()
+            $this->getCurrentCountry() === AbstractConsignment::CC_NL &&
+            $this->isMailboxActive() &&
+            $orderWeight &&
+            ($mailboxPercentage === 0 || $mailboxPercentage <= 100) &&
+            $orderWeight <= $maximumMailboxWeight
         ) {
-            return false;
+            $this->setMailboxPercentage($mailboxPercentage);
+
+            return true;
         }
 
-        return true;
+        return false;
     }
 
     /**
      * @return bool
      */
     public function fitInDigitalStamp(): bool
     {
+        $orderWeight               = $this->getWeightTypeOfOption($this->getWeight());
+        $maximumDigitalStampWeight = $this->getMaxDigitalStampWeight();
+
         if (
-            $this->getCurrentCountry() !== 'NL' ||
-            ! $this->isDigitalStampActive() ||
-            ! $this->isAllProductsFitInDigitalStamp() ||
-            $this->getWeight() > self::DEFAULT_DIGITAL_STAMP_WEIGHT
+            $this->getCurrentCountry() === AbstractConsignment::CC_NL &&
+            $this->isDigitalStampActive() &&
+            $orderWeight <= $maximumDigitalStampWeight
```

Comment:
Waarom heb je ervoor gekozen om hier variabelen te gebruiken en niet direct die functies?

Reply:
waarschijnlijk omdat ik anders had gezegd "dit is niet leesbaar, stop het in een variabele pls" :P het wordt dan ook een veel te lange regel